link-issue check fails on transient GitHub API errors — unguarded curl|jq in comment-maintenance steps

## Symptom

The required `Add Issue to PR / link-issue` check fails intermittently with:

```
jq: parse error: Invalid numeric literal at line 1, column 10
##[error]Process completed with exit code 5.
```

even when the PR is correctly linked (the run resolves `ISSUE_NUMBER` fine and the linked issue has a `Team :` label). Observed on PR #36606 (3 consecutive failures 2026-07-16 ~22:23–22:29 UTC, after 12 straight passes with identical linkage) and PR #36608 (failure 19:56, pass on retry 20:15). During the same window, direct `gh api` calls from a workstation intermittently returned HTML error bodies (`invalid character '<' looking for beginning of value`) with rate limits nowhere near exhausted — i.e. a transient GitHub API brownout.

## Root cause

`.github/workflows/issue_comp_link-issue-to-pr.yml` pipes raw `curl -s` output into `jq` in three steps with no HTTP-status or JSON-validity guard:

1. "Remove failure comment if resolved" — `existing_comments=$(curl -s …/comments)` → `echo … | jq -r`
2. "Remove issue-validation failure comment if resolved" — same pattern
3. "Get existing issue comments" — `curl -s …/comments | jq -c . > /tmp/issue_comments.json` (this is the step that failed on #36606: its env group ends with `ISSUE_NUMBER`, matching the crash location in the logs)

When the API returns a transient non-JSON error body (5xx HTML, secondary-rate-limit text), `jq` exits 5 and the `bash -e` step fails the whole required check. `curl -s` without `-f` also happily forwards HTTP error bodies as if they were data.

## Fix

Make the three fetches brownout-tolerant: `curl -sf --retry 3 --retry-all-errors`, and fall back to an empty JSON array (`[]`) if the fetch or parse still fails — the affected steps are best-effort comment maintenance, so the correct degraded behavior is "skip the cleanup / treat as no existing comments", never "fail the merge-gating check".

## Impact

Org-wide: any PR whose link-issue run coincides with a GitHub API hiccup gets a red required check and a confusing signal ("issue linking required" when linkage is fine). Retry usually passes, but authors have no way to know that.

--- a/.github/workflows/issue_comp_link-issue-to-pr.yml
+++ b/.github/workflows/issue_comp_link-issue-to-pr.yml
@@ -256,12 +256,15 @@ jobs:
           pr_url="$PR_URL"
           pr_number=$(echo "$pr_url" | grep -o '[0-9]*$')
 
-          # Check for existing failure comment
-          existing_comments=$(curl -s \
+          # Check for existing failure comment. Best-effort cleanup: a transient API
+          # error (5xx/HTML body) must degrade to "no comment found", never fail the
+          # merge-gating check by feeding non-JSON to jq (#36610).
+          existing_comments=$(curl -sf --retry 3 --retry-all-errors \
             -H "Accept: application/vnd.github+json" \
             -H "Authorization: Bearer ${{ env.GH_TOKEN }}" \
             -H "X-GitHub-Api-Version: 2022-11-28" \
-            "https://api.github.com/repos/${{ github.repository }}/issues/$pr_number/comments")
+            "https://api.github.com/repos/${{ github.repository }}/issues/$pr_number/comments") || existing_comments="[]"
+          echo "$existing_comments" | jq -e . >/dev/null 2>&1 || existing_comments="[]"
 
           # Find failure comment by looking for the distinctive header
           failure_comment_id=$(echo "$existing_comments" | jq -r '.[] | select(.user.login == "github-actions[bot]" and (.body | contains("❌ Issue Linking Required"))) | .id' | head -1)
@@ -338,11 +341,13 @@ jobs:
           pr_url="$PR_URL"
           pr_number=$(echo "$pr_url" | grep -o '[0-9]*$')
 
-          existing_comments=$(curl -s \
+          # Best-effort cleanup: transient API errors degrade to "no comment found" (#36610).
+          existing_comments=$(curl -sf --retry 3 --retry-all-errors \
             -H "Accept: application/vnd.github+json" \
             -H "Authorization: Bearer ${{ env.GH_TOKEN }}" \
             -H "X-GitHub-Api-Version: 2022-11-28" \
-            "https://api.github.com/repos/${{ github.repository }}/issues/$pr_number/comments")
+            "https://api.github.com/repos/${{ github.repository }}/issues/$pr_number/comments") || existing_comments="[]"
+          echo "$existing_comments" | jq -e . >/dev/null 2>&1 || existing_comments="[]"
           validation_comment_id=$(echo "$existing_comments" | jq -r '.[] | select(.user.login == "github-actions[bot]" and (.body | contains("❌ Linked Issue Needs Team Label"))) | .id' | head -1)
 
           if [[ -n "$validation_comment_id" && "$validation_comment_id" != "null" ]]; then
@@ -364,13 +369,17 @@ jobs:
         env:
           ISSUE_NUMBER: ${{ steps.determine_issue.outputs.final_issue_number }}
         run: |
-          # Fetch comments and write to temp file to avoid multiline JSON issues in GitHub Actions outputs
-          curl -s \
+          # Fetch comments and write to temp file to avoid multiline JSON issues in GitHub Actions outputs.
+          # A transient API error (5xx/HTML body) degrades to an empty list — downstream
+          # then posts a fresh comment instead of updating, rather than failing the
+          # merge-gating check on a jq parse error (#36610).
+          curl -sf --retry 3 --retry-all-errors \
               -H "Accept: application/vnd.github+json" \
               -H "Authorization: Bearer ${{ env.GH_TOKEN }}" \
               -H "X-GitHub-Api-Version: 2022-11-28" \
               "https://api.github.com/repos/${{ github.repository }}/issues/$ISSUE_NUMBER/comments" \
-            | jq -c . > /tmp/issue_comments.json
+            | jq -c . > /tmp/issue_comments.json \
+            || echo "[]" > /tmp/issue_comments.json
 
           echo "comments_file=/tmp/issue_comments.json" >> "$GITHUB_OUTPUT"
 

